nav2_behavior_tree/plugins/action/remove_passed_goals_action.cpp

```diff
@@ -57,16 +55,15 @@ inline BT::NodeStatus RemovePassedGoals::tick()
 
   using namespace nav2_util::geometry_utils;  // NOLINT
 
-  geometry_msgs::msg::PoseStamped current_pose;
-  if (!nav2_util::getCurrentPose(
-      current_pose, *tf_, global_frame_, robot_base_frame_,
-      transform_tolerance_))
-  {
-    return BT::NodeStatus::FAILURE;
-  }
-
   double dist_to_goal;
   while (goal_poses.size() > 1) {
+    geometry_msgs::msg::PoseStamped current_pose;
+    if (!nav2_util::getCurrentPose(
+        current_pose, *tf_, goal_poses[0].header.frame_id, robot_base_frame_,
```

Comment:
This can be moved back. The paths are all in the same frameID and you're pulling `[0]`s every time anyway.

Reply:
The first element with index 0 is a different one in each cycle, because the loop either terminates or the first element gets removed. But assuming that the path that is processed here is always the output of a planner, the assumption that the frame_id is the same for all poses is valid. I will move the call back up, as requested.

In general, I am not aware of a specification besides "best practice" and "that is how the interface is meant to be used" that paths could not be provided with a different `frame_id` for each `PoseStamped` instance? The [`nav_msgs/Path`](https://github.com/ros2/common_interfaces/blob/ad69eab30975b967b8c840c690ab491b2d6234cd/nav_msgs/msg/Path.msg) definition is not explicit with respect to restrictions on nested frame_ids and whether the timestamps per pose play a role or not. If they don't, and the `frame_id` must also match that of the message header in all cases, then the definition could have used unstamped `geometry_msgs/Pose[] poses` instead a long time ago... In my experience different tools and libraries use the message type in different ways, some don't even populate the individual pose headers. Related: https://github.com/ros2/common_interfaces/issues/97.

Reply:
The discussion you linked to I generally agreed that the `frame` was redundant and the `header` is mostly useful in the paths in case its a proper trajectory and each of the path poses have a timestamp attached to them - which could be made use of in the medium-term future.